src/main/java/net/snowflake/ingest/streaming/internal/ArrowRowBuffer.java

```diff
@@ -221,12 +221,18 @@ void setupSchema(List<ColumnMetadata> columns) {
    * synchronization
    */
   void close() {
-    if (this.vectorsRoot != null) {
-      this.vectorsRoot.close();
-      this.tempVectorsRoot.close();
-    }
+    // If the channel is invalid, cleanup all the leftover data; otherwise, simply close the
+    // allocator which will verify that everything should be cleaned up already
     this.fields.clear();
-    Utils.closeAllocator(this.allocator);
+    if (!this.owningChannel.isValid()) {
+      if (this.vectorsRoot != null) {
+        this.vectorsRoot.close();
```

Comment:
I do not know the details about what these `vectors.close()` do, but if it is needed, shouldn't we also call them during normal termination? Here we call them when `!owningChannel.isValid()` but what happens when the channel is valid?

Reply:
We don't really need them because the vectors should be cleaned up already doing normal termination (we always do a flush first), but I guess the logic looks confusing since @sfc-gh-mnaides had a similar question, I update the code to always do the cleanup, but checking the allocated memory should be already 0 before cleaning up for nomination :)